lib/omniauth/strategies/saml.rb

```diff
@@ -103,15 +109,23 @@ def other_phase
       uid { @name_id }
 
       info do
-        {
-          :name  => @attributes[:name],
-          :email => @attributes[:email] || @attributes[:mail],
-          :first_name => @attributes[:first_name] || @attributes[:firstname] || @attributes[:firstName],
-          :last_name => @attributes[:last_name] || @attributes[:lastname] || @attributes[:lastName]
-        }
+        found_attributes = options.attribute_statements.map do |key, values|
+          attribute = find_attribute_by(values)
+          [key, attribute]
```

Comment:
Does it make sense to have `[key, nil]` here if no mapped attribute is found? I guess that's how the old code worked, so it's probably fine.